src/sidebar/services/router.js

```diff
@@ -81,43 +94,39 @@ export default function router($window, store) {
     return url;
   }
 
-  let didRegisterPopstateListener = false;
-
   /**
    * Synchronize the route name and parameters in the store with the current
    * URL.
    *
    * The first call to this method also registers a listener for future back/forwards
    * navigation in the browser.
    */
-  function sync() {
-    const { route, params } = currentRoute();
-    store.changeRoute(route, params);
+  sync() {
+    const { route, params } = this.currentRoute();
+    this._store.changeRoute(route, params);
 
     // Set up listener for back/forward navigation. We do this in `sync()` to
     // avoid the route being changed by a "popstate" emitted by the browser on
     // document load (which Safari and Chrome do).
-    if (!didRegisterPopstateListener) {
-      $window.addEventListener('popstate', () => {
+    if (!this._didRegisterPopstateListener) {
+      this._window.addEventListener('popstate', () => {
         // All the state we need to update the route is contained in the URL, which
         // has already been updated at this point, so just sync the store route
         // to match the URL.
-        sync();
+        this.sync();
       });
-      didRegisterPopstateListener = true;
+      this._didRegisterPopstateListener = true;
     }
   }
 
   /**
    * Navigate to a given route.
    *
    * @param {string} name
-   * @param {Object} params
+   * @param {Object.<string,string>} params
```

Comment:
```suggestion
   * @param {Record<string, string>} params
```

Reply:
Or maybe the type suggested above.

Reply:
We're currently using a mix of `Record<Key, Value>` and `Object.<Key,Value>` in the code. We had a discussion about this a while back and you mentioned that [`Record` is stricter](https://hypothes-is.slack.com/archives/C1M8NH76X/p1611230893009700?thread_ts=1611226352.008200&cid=C1M8NH76X) which could help to avoid some errors, so I think I'm good with this change going forwards.